### PR TITLE
[10.0][IMP] partner_identification: Add field computation and inverses

### PR DESCRIPTION
**Pull request by lasley**
This PR add methods to allow for computation and inverse of an ID field of a specific category type, such as in the following example:

``` python
social_security_id = fields.Char(
    string='Social Security',
    compute=lambda s: s._compute_identification(
        'social_security_id', 'SSN',
    ),
    inverse=lambda s: s._inverse_identification(
        'social_security_id', 'SSN',
    ),
)
```

In the above example, the first Identification found in the `SSN` category code for the partner will be used for computation. The inverse blocks write when there is more than one active ID for the same type and partner in order to avoid accidental mutation of an incorrect record.

This has been working in Vertical Medical core for a while now, I'd just like to move it here for more abstract usage.

### CODE REVIEW

**Comment by lasley** [comment]
cc @jbeficent - this was moved from OCA/vertical-medical#174

**Comment by lmignon** [comment]
@lasley Thank you for this new feature.  The code LGTM. It would be nice to also provide search method to be complete. Nevertheless :+1: (Code review only)

**Review by lmignon — APPROVED** [review]

**Review by luismontalba — APPROVED** [review]
Functionality tested on runbot

**Comment by rafaelbn** [comment]
Please @lasley could you review travis errors? Thanks!

**Review by cubells — APPROVED** [review]
:+1: 
LGTM

**Comment by cubells** [comment]
Travis error not related to module.

**Comment by pedrobaeza** [comment]
@lasley do you know why the branch is red?

**Review by AaronHForgeFlow — COMMENTED** [review]
It works good, pending to figure it out what is causing travis to fail

**Comment by rafaelbn** [comment]
Well, @lasley it could be nice to discover why the branch is red, in any case I don't think we must block merging this PR in any case

**Comment by rafaelbn** [comment]
Well, one doubt, in you first commit it was no error, in the second it appears.

**Review by rafaelbn — APPROVED** [review]
Travis says red but there is no error in https://travis-ci.org/OCA/partner-contact/jobs/236469721 so is 👍

**Comment by lasley** [comment]
Dang now the 10.0 branch is failing due to the merge. I don't see an error anywhere though...

**Comment by yajo** [comment]
Is it possible that it's due to some timeout? Travis is under heavy load with the #ocacodesprint.

**Comment by lasley** [comment]
Hmmmmm possible. We'll know soon enough!

**Comment by yajo** [comment]
It's https://github.com/OCA/maintainer-quality-tools/issues/448 definitely.

**Pull request merged**
